restAssuredReqres/src/test/java/api/ReqresTests.java

```diff
@@ -1,166 +1,166 @@
-        register.put("password", "");
-        Services.postRequest("api/register", register)
-            .then()
-            .body("error", equalTo("Missing password"));
-    }
-
-    @Test(description = "LOGIN - SUCCESSFUL")
-    public void checkSuccessfulLoginTest() {
-        Specifications.installSpecifications(Specifications.requestSpec(URL),
-            Specifications.responseSpec(StatusCode.OK.getCode()));
-        register.put("email", "eve.holt@reqres.in");
-        register.put("password", "cityslicka");
-        Services.postRequest("api/login", register)
-            .then()
-            .body("token", equalTo("QpwL5tke4Pnpja7X4"));
-    }
-
-    @Test(description = "LOGIN - UNSUCCESSFUL")
-    public void checkUnsuccessfulLoginTest() {
-        Specifications.installSpecifications(Specifications.requestSpec(URL),
-            Specifications.responseSpec(StatusCode.BAD_REQUEST.getCode()));
-        register.put("email", "peter@klaven");
-        register.put("password", "");
-        Services.postRequest("api/login", register)
-            .then()
-            .body("error", equalTo("Missing password"));
-    }
-
-    @Test(description = "DELAYED RESPONSE")
-    public void getDelayedListUsersTest() {
-        Specifications.installSpecifications(Specifications.requestSpec(URL),
-            Specifications.responseSpec(StatusCode.OK.getCode()));
-        Services.getRequest("/api/users?delay=3")
-            .then()
-            .body("data.id[0]", equalTo(1));
-    }
+package api;
+
+import enums.StatusCode;
+import java.util.HashMap;
+import java.util.Map;
+import org.testng.annotations.Test;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.notNullValue;
+
+
+public class ReqresTests {
+
+    private final static String URL = "https://reqres.in/";
+    private final Map<String, String> user = new HashMap<>();
+    private final Map<String, String> register = new HashMap<>();
+
+    @Test(description = "LIST USERS")
+    public void getListUsersTest() {
+        Specifications.installSpecifications(Specifications.requestSpec(URL),
+            Specifications.responseSpec(StatusCode.OK.getCode()));
+        Services.getRequest("api/users?page=2")
+            .then()
+            .body("data.id[0]", equalTo(7));
+    }
+
+    @Test(description = "SINGLE USER")
+    public void getSingleUserTest() {
+        Specifications.installSpecifications(Specifications.requestSpec(URL),
+            Specifications.responseSpec(StatusCode.OK.getCode()));
+        Services.getRequest("api/users/2")
+            .then()
+            .body("data.id", equalTo(2));
+    }
+
+    @Test(description = "SINGLE USER NOT FOUND")
+    public void getNonExistentUserTest() {
+        Specifications.installSpecifications(Specifications.requestSpec(URL),
+            Specifications.responseSpec(StatusCode.NOT_FOUND.getCode()));
+        Services.getRequest("api/users/23");
+    }

```

Comment:
Not sure what do you verify at this test. Your only verification in the Specifications.responseSpec(StatusCode.NOT_FOUND.getCode())? Looks a little bit strange for me, but maybe because for now there is not assertThat

Reply:
Maybe we have some response body and we can verify it?